metricbeat/mb/registry.go

```diff
@@ -300,11 +300,33 @@ func (r *Register) Modules() []string {
 	r.lock.RLock()
 	defer r.lock.RUnlock()
 
+	var dups = map[string]int{}
+
+	//for the sake of compatibility, grab modules the old way as well, right from the modules map
 	modules := make([]string, 0, len(r.modules))
 	for module := range r.modules {
+		dups[module]++
 		modules = append(modules, module)
 	}
 
+	// List also modules from secondary sources
+	if source := r.secondarySource; source != nil {
+		sourceModules, err := source.Modules()
+		if err != nil {
+			logp.Error(errors.Wrap(err, "failed to get modules from secondary source"))
+		}
+		modules = append(modules, sourceModules...)
+	}
+
+	//grab a more comprehensive list from the metricset keys, then reduce and merge
```

Comment:
Keep a consistent style for comments.
```suggestion
	// Grab a more comprehensive list from the metricset keys, then reduce and merge
```